app/controllers/schools_controller.rb

```diff
@@ -34,9 +34,79 @@ def admins
     authorize current_school
   end
 
+  # GET /school/standing
+  def standing
+    authorize current_school
+    @presenter = Schools::StandingPresenter.new(view_context, current_school)
+  end
+
+  # PATCH /school/standing
+  def toggle_standing
+    authorize current_school
+
+    standing_enabled = params[:enable_standing] == "true"
+
+    Standing.transaction do
+      current_school.update!(
+        configuration:
+          current_school.configuration.merge(
+            "enable_standing" => standing_enabled
+          )
+      )
+
+      if standing_enabled && current_school.standings.count.zero?
```

Comment:
>  Can't the user create a standing from the UI?

The user can create the standings from the UI. 

> Why do we need to do this?

When enabling the standing for the first time, it will create a default standing. It will be the default standing for the users in the school.

The default standing cannot be deleted. It can be edited and updated. 

All the other standings created by admin users cannot be made as default. All the other fields can be edited.

Reply:
Why do we need to do that? Can't we check for the presence of standing instead of caching the same with a flag?

Reply:
> Why do we need to do that? Can't we check for the presence of standing instead of caching the same with a flag?

Can you please elaborate?

Reply:
Currently we are creating default standing when standing is enabled;

Why not allow users to create standing when they want and standing will only be enabled when they actually create one; 🤔

Reply:
Enabling standing itself is an indication that they would like to enable standing for the school. The UI clearly mentions `Enable standing for the school ?` `Yes` or `No`

Enabling the standing only when a user creates a standing record, then the whole purpose of the above step will not make sense.

They can still edit and update the default.

By creating a default standing, it made some part of the code simple. Like not showing the default option in form fields, and also it streamlines the user experience.

It provides a practical reference point, 

we've decided to continue with the approach of automatically creating a default standing when the feature is enabled.

----

I acknowledge that the approach you suggested is a valid alternative, offering a different way to handle the feature.

Reply:
> Enabling the standing only when a user creates a standing record, then the whole purpose of the above step will not make sense.

Yep, even I was thinking that step was redundant; That flag on the School config doesn't serve any purpose, even if its enabled you still need a default standing record for the system to work; So why not consider standing enabled when the user creates the 1st standing; 🤔

Reply:
> That flag on the School config doesn't serve any purpose, even if its enabled you still need a default standing record for the system to work; So why not consider standing enabled when the user creates the 1st standing;

@bodhish IIRC, the reasoning was two-fold.

1. Clear UX - this allows the user to toggle the feature on and off without affecting the stored standings. 
2. Easy querying - To determine whether we should show the standings link in a user's dashboard, only the configuration flag needs to be checked in `current_school` (probably already loaded at the time of check)... as opposed to... the presence of any _live_ (unarchived) standings in the school.

While folding this down to rely only on standings entry is possible, that feels like _conflation of purpose_ / overengineering to me.